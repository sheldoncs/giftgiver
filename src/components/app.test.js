import React from "react";
import { shallow, configure } from "enzyme";
import App from "./App";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("App", () => {
  const app = shallow(<App />);
  const id = 1;
  it("Renders Corectly", () => {
    expect(app).toMatchSnapshot();
  });
  it("Initializes `state` with 'empty list of gifts'", () => {
    expect(app.state().gifts).toEqual([]);
  });

  describe("when clicking `Add Gift` button ", () => {
    beforeEach(() => {
      app.find(".btn-add").simulate("click");
    });
    afterEach(() => {
      app.setState({ gifts: [] });
    });
    it("Adds new gift to `state`", () => {
      expect(app.state().gifts).toEqual([{ id: 1 }]);
    });
    it("Adds a new gift to the rendered list ", () => {
      expect(app.find(".gift-list").children().length).toEqual(1);
    });
    it("creates a gift component", () => {
      expect(app.find("Gift").exists()).toBe(true);
    });
  });
  describe("when user wants to remove the added gift", () => {
    beforeEach(() => {
      app.instance().removeGift(id);
    });
    it("removes gift from `state` ", () => {
      expect(app.state().gifts).toEqual([]);
    });
  });
});
