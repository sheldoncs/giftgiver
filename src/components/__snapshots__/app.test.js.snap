// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Corectly 1`] = `ShallowWrapper {}`;
